### PR TITLE
Lesson 10 Content

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: src/chapters/razor-views/template-forms.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Start a New Project
+-------------------
+
+You will build a new project so you can practice with templates and forms.
+If you have not done so, commit and push any unsaved work from your ``HelloASPDotNET`` project.
+
+.. TODO: Add link back to setup steps
+
+Your new project will keep track of some fictional coding events.
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/template-forms.rst]: Suggestion: "...some coding events, such as meetups and conferences."

File: src/chapters/razor-views/template-forms.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+To get started, follow the steps you took to create ``HelloASPDotNET``, but call the project ``CodingEvents``.
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/template-forms.rst]: Is there a previous page or video that we can link to here?

File: src/chapters/razor-views/template-forms.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+You can include as many inputs as you need in the form, and these can be of
+different types (e.g. text, email, checkbox, etc.). However, each different
+piece of data you want to collect needs to have a unique ``name`` attribute.
+
+To *render* the form in the view, add a method to the controller using the
+``[HttpGet]`` annotation:
+
+.. sourcecode:: csharp
+   :linenos:
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/template-forms.rst]: Do we have the finished video/lesson code branch? If so, making the line numbers match up with that code would be good. (Maybe you are already doing this as part of the process...if so, please ignore!)

File: src/chapters/razor-views/template-forms.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+As usual, the following summary outlines the ideas from the clip.
+
+Add a Form Handler Method - Text
+--------------------------------
+
+To *process* a form after the user clicks the *Submit* button, you need to add
+a method to the controller using the ``[HttpPost]`` annotation:
+
+.. sourcecode:: csharp
+   :linenos:
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/template-forms.rst]: Same thing as above regarding line numbers.

File: src/chapters/razor-views/template-forms.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      return View();
+   }
+
+Some points to note:
+
+#. Line 2: For each piece of data that needs to be retrieved from the form,
+   declare a parameter of the appropriate type.
+#. The method code performs any data manipulation required after the
+   information gets submitted.
+#. Line 6: Generally, we want to send the user to a different page after they
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/template-forms.rst]: Is this accurate, given the code above? (My .NET is rusty...)

File: src/chapters/razor-views/layout-pages.rst [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+The header and footer of our site when we first create it contains links to important info like a privacy statement.
+As you work on your application, you may want your header and footer for each page to include your own privacy statement or a contact page.
+When you make updates to ``_Layout``, every template in the application using ``_Layout`` reflects the updates you have made.  
+
+.. index:: ! partial view
+
+Partial Views
+-------------
+
+If you want to create repeatable elements of your webpage that are contained within the main body of the page, you can use a **partial view(+**.
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/layout-pages.rst]: "partial view(+" is a typo?

File: src/chapters/razor-views/layout-pages.rst [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+Whenever we create a new project, we get one layout page, ``_Layout.cshtml``, and this file contains links to Bootstrap.
+Layout pages in ASP.NET MVC focus on the header and footer of the site. 
+The header and footer of our site when we first create it contains links to important info like a privacy statement.
+As you work on your application, you may want your header and footer for each page to include your own privacy statement or a contact page.
+When you make updates to ``_Layout``, every template in the application using ``_Layout`` reflects the updates you have made.  
+
+.. index:: ! partial view
+
+Partial Views
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/layout-pages.rst:21]: The definition of partial views is solid, but I think this section could benefit from a more explicit example. For example, the periodic table example referenced in the first section of this chapter would work well!

File: src/chapters/razor-views/layout-pages.rst [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+-------------
+
+If you want to create repeatable elements of your webpage that are contained within the main body of the page, you can use a **partial view(+**.
+Partial views are blocks of HTML elements that we want to use across multiple templates.
+When creating a partial view, we add the file to the ``Shared`` directory.
+Then we can upon the patial view in the appropriate spot in our template by using ``Render()``.
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/layout-pages.rst]: When calling `Render()` is it necessary to pass in the name of the partial?

File: src/chapters/razor-views/static-resources.rst [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+While every view we add for our action methods in the controllers inherits from this template, we can also add view-specific styles and scripts.
+
+To add styles that only apply to the view we are currently working on, we can use the same ``<link>`` tag we normally do for a static HTML page.
+We can also add JS files the same way. 
+However, we want to make sure of a couple things first:
+
+#. That we are overriding any styles and scripts from ``_Layout``.
+#. That we have put custom CSS or JS in a separate ``Content`` folder.
+
+We can also put any images that we need in the separate ``Content`` folder as well.
```
Comment (chrisbaylewis) [inline src/chapters/razor-views/static-resources.rst]: To your question (via Slack), I don't think this section requires a video. 

My only suggestion would be to add a sentence or two explaining _why_ static stuff has to go in `Content/` and how paths work in this context. For example, if we have the file `Content/mystyles.css` can we use `<link href="mystyles.css">` or is another path required?

**Review by chrisbay — APPROVED** [review]
Approved with minor suggested changes.

File: src/chapters/razor-views/template-forms.rst (hunk 1 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/template-forms.rst]: Done! Thanks!

File: src/chapters/razor-views/template-forms.rst (hunk 2 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/template-forms.rst]: Done!

File: src/chapters/razor-views/layout-pages.rst (hunk 6 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/layout-pages.rst]: Fixed!

File: src/chapters/razor-views/template-forms.rst (hunk 3 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/template-forms.rst]: We do! Double-checked the line numbers and they match!

File: src/chapters/razor-views/template-forms.rst (hunk 4 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/template-forms.rst]: Done!

File: src/chapters/razor-views/template-forms.rst (hunk 5 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/template-forms.rst]: Yup! We are using ``Redirect()`` here to send the users back to the ``Index`` template. I added a little list following the item submitted through the different steps to being displayed on the ``Index`` page so I think that will help clarify as well.

File: src/chapters/razor-views/layout-pages.rst (hunk 7 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/layout-pages.rst:21]: Done!

File: src/chapters/razor-views/layout-pages.rst (hunk 8 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/layout-pages.rst]: It is! I did find an easier way to do it with an HTML tag, so I subbed the tag for ``Render()``.

File: src/chapters/razor-views/static-resources.rst (hunk 9 above)
Comment (gildedgardenia) [reply src/chapters/razor-views/static-resources.rst]: Fixed!

**Pull request merged**